test/functional/uri_options_spec_tests.js

```diff
@@ -22,8 +23,14 @@ describe('URI Options (spec)', function() {
             metadata: { requires: { topology: 'single' } },
             test: function(done) {
               parse(test.uri, {}, (err, result) => {
-                expect(err).to.not.exist;
-                expect(result.options).to.containSubset(test.options);
+                if (test.valid === true) {
+                  expect(err).to.not.exist;
+                  expect(result.options).to.containSubset(test.options);
+                } else {
+                  console.log(err);
+                  expect(err).to.exist;
```

Comment:
Nit: this assertion is mostly covered by the next one. The error message if this happens is going to look like `expected null to exist`, which is not super helpful.

Reply:
Good point